presto-parser/src/main/java/com/facebook/presto/sql/tree/Identifier.java

```diff
@@ -87,12 +87,12 @@ public boolean equals(Object o)
         }
 
         Identifier that = (Identifier) o;
-        return Objects.equals(value, that.value);
```

Comment:
Whether they are the same or not depends on the context. i.e. in a SelectItem, un-delimited identifier matches column name in an case-insensitive way (`a` matches with both `a` and `A`, while `"a"` only matches `a`); in alias, however, both identifier have the same meaning as long as their values are the same. Thus, for 2 identifiers to be absolutely equivalent in all context, delimited should be compared. @martint